common/ts/publish-task.ts

```diff
@@ -28,13 +31,30 @@ export default async function publishTask(endpointType: EndpointType) {
     taskReports.map(taskReport => getReportForTask(taskReport, metrics, endpoint, timeoutSec))
   );
 
+  if (globalQualityGateStatus === '') {
+    globalQualityGateStatus = 'ok';
```

Comment:
There could be multiple analyses, and multiple task reports couldn't there? Won't the last task to be reported "win" in that case?

Reply:
The algorithm works like this : If one of the quality gate from the multiple analyses has an error / warn / none status then the global quality gate will fail, and that will not change if an 'ok' one appear after. To have the global status green, all of the analyses should have the 'OK' status, that's the reason of this code (variable not modified, so it's ok), see line 74/75/76 for the tests.